src/Compute/v2/Api.php

```diff
@@ -399,4 +427,132 @@ public function deleteServerMetadataKey(): array
             ]
         ];
     }
+
+    public function getKeypair(): array
+    {
+        return [
+            'method' => 'GET',
+            'path'   => 'os-keypairs/{name}',
+            'params' => [
+                'name' => $this->isRequired($this->params->keypairName())
+            ],
+        ];
+    }
+
+    public function getKeypairs(): array
+    {
+        return [
+            'method' => 'GET',
+            'path'   => 'os-keypairs',
+            'params' => [],
+        ];
+    }
+
+    public function postKeypair(): array
+    {
+        return [
+            'method' => 'POST',
+            'path'   => 'os-keypairs',
+            'jsonKey' => 'keypair',
+            'params' => [
+                'name'  => $this->isRequired($this->params->name('keypair')),
+                'publicKey' => $this->params->keypairPublicKey(),
+            ]
+        ];
+    }
+
+    public function deleteKeypair(): array
+    {
+        return [
+            'method' => 'DELETE',
+            'path'   => 'os-keypairs/{name}',
+            'params' => [
+                'name' => $this->isRequired($this->params->keypairName())
+            ]
+        ];
+    }
+
+    public function addSecurityGroup(): array
```

Comment:
This method is to attach a security group to a server instance. `postSecurityGroups` creates new SG.